app/services/metrics/metrics_processor.rb

```diff
@@ -0,0 +1,73 @@
+module Metrics
+  ##
+  # Does the processing of the Github events previously stored and leaves the
+  # generated metrics in the Metrics table.
+  #
+  # It takes care of processing only the metrics that are pending to be
+  # processed.
+  class MetricsProcessor < BaseService
+    ##
+    # Processes the metrics
+    def call
+      process_all_metrics
+    end
+
+    private
+
+    ##
+    # Returns all the defined metrics to process.
+    def all_metrics_definitions
+      MetricsDefinition.all
+    end
+
+    ##
+    # Returns the concrete MetricsProcessor to process the metrics definition.
+    def metrics_processor_for(metrics_definition)
+      metrics_definition.metrics_processor.constantize
+    end
+
+    ##
+    # Makes each metric defined to process all the events.
+    def process_all_metrics
+      metrics_definitions = all_metrics_definitions
+
+      events = events_to_process_for(metrics_definitions)
+
+      return if events.empty?
+
+      metrics_definitions.each do |metrics_definition|
+        metrics_processor = metrics_processor_for(metrics_definition)
+
+        process(metrics_processor: metrics_processor,
+                events: events,
+                metrics_definition: metrics_definition)
+      end
+    end
+
+    ##
+    # Makes the given metric to process all the events.
+    def process(metrics_processor:, events:, metrics_definition:)
+      time_interval = metrics_definition.time_interval_starting_at(Time.zone.today)
+
+      metrics_processor.call(events: events, time_interval: time_interval)
+
+      metrics_definition.update!(last_processed_event_time: events.last.created_at)
+    end
+
+    ##
+    # Returns an array with the events to process.
+    # It polls the given metrics_definitions to know which events to query
+    # to avoid querying more events than needed.
+    def events_to_process_for(metrics_definitions)
+      Event.received_after(minimum_time_among(metrics_definitions))
+           .order(:created_at)
```

Comment:
is created_at enough? I mean with webhooks and jobs (with retries) I'm not sure if the order of creation in our DB will always match the order in which the events actually happened